packages/aws-cdk-lib/aws-cloudwatch-actions/lib/lambda.ts

```diff
@@ -23,12 +23,17 @@ export class LambdaAction implements cloudwatch.IAlarmAction {
    */
   bind(scope: Construct, alarm: cloudwatch.IAlarm): cloudwatch.AlarmActionConfig {
     const idPrefix = FeatureFlags.of(scope).isEnabled(LAMBDA_PERMISSION_LOGICAL_ID_FOR_LAMBDA_ACTION) ? alarm.node.id : '';
-    this.lambdaFunction.addPermission(`${idPrefix}AlarmPermission`, {
-      sourceAccount: Stack.of(scope).account,
-      action: 'lambda:InvokeFunction',
-      sourceArn: alarm.alarmArn,
-      principal: new iam.ServicePrincipal('lambda.alarms.cloudwatch.amazonaws.com'),
-    });
+    const permissionId = `${idPrefix}AlarmPermission`;
+    const permissionNode = this.lambdaFunction.permissionsNode.tryFindChild(permissionId) as lambda.CfnPermission | undefined;
+
+    if (permissionNode?.sourceArn !== alarm.alarmArn) {
```

Comment:
```suggestion
    // If the Lambda permission has already been added to this function
    // and the feature flag is disabled we skip adding it to avoid an exception being thrown
    // see https://github.com/aws/aws-cdk/issues/29514
    if (permissionNode?.sourceArn !== alarm.alarmArn) {
```

Reply:
Added the comment but modified it because we will avoid the exception with the feature flag enabled or disabled.

Reply:
Hey ✋ Sorry for the late reply.
I wanted to go through it a bit more to clarify some doubts.

I think that what we wanna do here is keep the behavior consistent when not under FF.
Ie, it should throw with these scenarios:
1. Same alarm, different action, same lambda
```ts
alarm1.addAlarmAction(new actions.LambdaAction(alarmLambda));
alarm1.addOkAction(new actions.LambdaAction(alarmLambda));
```
2. Same alarm, same action, same lambda
```ts
alarm.addAlarmAction(new actions.LambdaAction(alarmLambda));
alarm.addAlarmAction(new actions.LambdaAction(alarmLambda));
```
3. Different alarm, different action, same lambda
```ts
alarm1.addAlarmAction(new actions.LambdaAction(alarmLambda));
alarm2.addAlarmAction(new actions.LambdaAction(alarmLambda));
```
That's because we try to create the same `AlarmPermission` construct.

With FF enabled we should address cases 1 and 2 (since 3 is already handled), that's why I suggested we should use the condition from the first commit as the other approach would be adding a second FF to add another prefix to the construct name (seems cumbersome).

Let me know if you have any doubts or comments.

Thanks.

Reply:
Hey ✋
Sorry for the late response, needed a bit of free time to go through it.
Anyway, thanks for the changes. 
I think you made valid points with your responses and the PR looks good to me 👍

Reply:
No worries! Appreciate you taking the time to go through it all and thank you for the feedback.